### PR TITLE
es6 and commonjs versions of packages

### PR DESCRIPTION
**Pull request by adrinr**
## Description
Creates an es6 and a commonjs version of the types and the shared-core. Required to allow the consumption of both frontend and backend services, via node imports, rollup, vite and jest.

### CODE REVIEW

File: packages/client/rollup.config.js [hunk 1]
```diff
@@ -20,6 +20,19 @@ const ignoredWarnings = [
   "a11y-click-events-have-key-events",
 ]
 
+const devPaths = production
+  ? []
+  : [
+      {
+        find: "@budibase/shared-core",
+        replacement: path.resolve("../shared-core/dist/mjs/src/index"),
```
Comment (aptkingston) [inline packages/client/rollup.config.js:28]: Is this still needed, even with both `shared-core` and `types` now specifying a default ESM `import` export? The client should be able to consume it now without these additional aliases, like it does for `string-templates` which also gets built to both CJS and ESM bundles.

File: packages/shared-core/package.json [hunk 2]
```diff
@@ -2,21 +2,28 @@
   "name": "@budibase/shared-core",
   "version": "2.4.8-alpha.1",
   "description": "Shared data utils",
-  "main": "dist/src/index.js",
-  "module": "dist/src/index.js",
-  "types": "dist/src/index.d.ts",
+  "main": "dist/mjs/src/index.js",
```
Comment (aptkingston) [inline packages/shared-core/package.json:5]: Normally I would say that the `main` key should probably point to the CJS build, but when this module is intended to be consumed via Typescript from node environments maybe that changes things.

**Review by aptkingston — APPROVED** [review]
LGTM, but I think we can remove the client aliases :+1:

File: packages/client/rollup.config.js (hunk 1 above)
Comment (adrinr) [reply packages/client/rollup.config.js:28]: It's needed for the local dev for now, as the published package does not have it, and rollup does not know about the local package.

Comment (adrinr) [reply packages/client/rollup.config.js:28]: This only runs on development, for production it will always read from the npm repo as it does now

**Pull request merged**